Improve dark mode styles across multiple pages

menaguilherme opened the pull request:
While using the app in dark mode, I noticed that in some places the text was hard to read or not visible at all due to low contrast.

This PR adjusts the styles on several pages to improve readability — mostly by tweaking background and text colors where the contrast was too low. I tried to stick to the existing theme and keep things consistent.

Let me know if you'd like any changes!
Also, let me know if you’d like a short video demo — happy to record one if it helps!

neo773 commented:
@menaguilherme 
Updated patch for more fixes

menaguilherme commented:
@neo773 Thank you for taking the time to update the patch — I’ve applied the changes and pushed them!

zachgoll commented on app/assets/tailwind/maybe-design-system.css:
```diff
@@ -788,4 +848,20 @@
+
+  @variant theme-dark {
+    @apply bg-black ;
+  }
```
We'll want to use our Figma tokens here (see doc I posted)

zachgoll commented on app/views/accounts/_account_type.html.erb:
```diff
@@ -1,7 +1,7 @@
 
 <%= link_to new_polymorphic_path(accountable, step: "method_select", return_to: params[:return_to]),
-            class: "flex items-center gap-4 w-full text-center focus:outline-hidden focus:bg-alpha-black-25 hover:bg-alpha-black-25 border border-transparent block px-2 rounded-lg p-2" do %>
+            class: "flex items-center gap-4 w-full text-center focus:outline-hidden focus:bg-alpha-black-25 hover:bg-alpha-black-25 dark:hover:bg-overlay dark:focus:bg-overlay fg-contrast hover:fg-primary focus:fg-primary border border-transparent block px-2 rounded-lg p-2" do %>
```
We'll want to rely on our centralized `maybe-design-system.css` to define these hover/focus states in each "mode" to avoid the pseudo selectors `dark:*` in the views.

zachgoll commented on app/views/categories/_form.html.erb:
```diff
@@ -61,10 +61,10 @@
+        <%= f.text_field :name, placeholder: t(".placeholder"), required: true, autofocus: true, label: "Name", data: { color_avatar_target: "name" }, class: "form-field__input text-ellipsis text-subdued" %>
         <% unless category.parent? %>
-          <%= f.select :parent_id, categories.pluck(:name, :id), { include_blank: "(unassigned)", label: "Parent category (optional)" }, disabled: category.parent?, data: { action: "change->category#handleParentChange" } %>
+          <%= f.select :parent_id, categories.pluck(:name, :id), { include_blank: "(unassigned)", label: "Parent category (optional)" }, disabled: category.parent?, data: { action: "change->category#handleParentChange" }, class: "form-field__input text-ellipsis text-subdued" %>
```
These selects/inputs should already be receiving styles from our centralized form builder, so we can update things there to centralize everything.

zachgoll reviewed (commented):
Thanks for these fixes, appreciate it!!

We're very much a WIP when it comes to design system, but I've published a little reference here that might help with this PR:

https://github.com/maybe-finance/maybe/wiki/Design-System

There are 2 main areas that I think we need to address (have left some contextual comments throughout):

- We should try to consolidate all form related styles into our form builder (see doc above)
- Remove all usage of `dark:some-style` (see doc above)

Other than those two points this looks good!

neo773 commented on app/views/layouts/wizard.html.erb:
```diff
@@ -1,5 +1,5 @@
 <%= render "layouts/shared/htmldoc" do %>
-  <div class="bg-primary-adaptive flex flex-col h-dvh pt-safe">
+  <div class="bg-container-inset flex flex-col h-dvh pt-safe">
```
this is not the right background color it conflicts with primary button, needs to be pure black

menaguilherme replied:
Thanks for pointing that out! I’ll revert that part and bring back the original background to avoid any conflicts. Appreciate the heads up!

menaguilherme commented:
@zachgoll All feedback addressed — removed all `dark:` usages, replaced them with design tokens, and updated `number_field` and `date_field` to use the `StyledFormBuilder`.

Still a bit unsure how to best handle the `bg-divider-adaptive` cases — I reintroduced the utility in one spot where the token replacement didn’t quite work visually. Open to suggestions if there’s a better direction.

Appreciate the guidance!

zachgoll commented on app/helpers/styled_form_builder.rb:
```diff
@@ -59,6 +63,30 @@ def submit(value = nil, options = {})
+    field = super(method, merged_options)
+
+    build_styled_field(label, field, merged_options)
+  end
```
I think these can move up to the metaprogramming block we have here, as both number/date fields are captured there:

https://github.com/maybe-finance/maybe/blob/a7dfafc9072b2986834a0b9fff60e8241b1d82f1/app/helpers/styled_form_builder.rb#L6-L16

zachgoll commented on app/views/rule/actions/_action.html.erb:
```diff
@@ -9,13 +9,13 @@
                 data: { rule__actions_target: "actionValue" } do %>
       <span class="font-medium uppercase text-xs">to</span>
-      <%= form.select :value, action.options || [], {}, disabled: !needs_value %>
+      <%= form.select :value, action.options || [], {}, class: "form-field__input text-primary", disabled: !needs_value %>
```
I _think_ this needs to be removed? This form should be using `styled_form_with`.  Unless there was a specific reason to override that I'm not seeing?

zachgoll commented on app/views/rule/conditions/_condition.html.erb:
```diff
@@ -14,19 +14,19 @@
+          <%= form.number_field :value, placeholder: "10", step: 0.01, class: "form-field__input text-primary" %>
         <% else %>
-          <%= form.text_field :value, placeholder: "Enter a value" %>
+          <%= form.text_field :value, placeholder: "Enter a value", class: "form-field__input text-primary" %>
```
Similar to comment above, think we can leverage the global form builder styles here?

zachgoll reviewed (commented):
Looking good!

Just a few details to finish things up:

- I'm still seeing a few `class: "..."` overrides in a few forms that I _think_ we can remove and just leverage our form builder styles for
- Address comment within form builder itself

In regards to `bg-divider-adaptive`, just spoke with @justinfar and he said for "dividers", we should be using those "border" utilities.

In this case, since we have `div` elements representing dividers, I think the most natural way to implement this is to switch those over to `<hr>` elements with that border.  Does that work?

```erb
<hr class="border border-secondary" />
```

menaguilherme commented:
@zachgoll  All feedback addressed — removed redundant `class:` usage from form fields, moved `number_field` and `date_field` into the builder’s metaprogramming block, and replaced the `bg-divider-adaptive` dividers with semantic `<hr>` elements using `border-secondary`.

Apologies for the extra back-and-forth — I missed a few things earlier but really appreciate the clear direction and your patience throughout. Let me know if there’s anything else I should adjust!

zachgoll commented on app/helpers/styled_form_builder.rb:
Sorry to backtrack on this, but after seeing the final changes I don't think we need to make any changes to this file at all.  Let's just make sure `form-field__input` has all the classes needed in the main design system CSS file and we should be good to go.

That way, whether we're using a form builder or manually passing in `form-field__input`, we'll get the same styling result.

So in summary:

- Revert all changes to this file
- Update main CSS file to ensure it has the `text-ellipsis placeholder:text-subdued`

zachgoll commented on app/views/layouts/wizard.html.erb:
```diff
@@ -1,5 +1,5 @@
 <%= render "layouts/shared/htmldoc" do %>
-  <div class="flex flex-col h-dvh pt-safe">
+  <div class="bg-primary-adaptive flex flex-col h-dvh pt-safe">
```
```suggestion
  <div class="bg-container flex flex-col h-dvh pt-safe">
```

menaguilherme replied:
Hey @zachgoll , I applied the bg-container change and it works well overall, but I noticed some contrast issues in dark mode, especially for layered components like this budget setup step.

Here’s a comparison between light and dark mode with the current tokens applied:

📸 Light mode: 
![Screenshot 2025-04-21 at 18 14 04](https://github.com/user-attachments/assets/8d8f4542-a99d-48d0-b15d-25a1ba0e4f19)

🌒 Dark mode: 
![Screenshot 2025-04-21 at 18 13 58](https://github.com/user-attachments/assets/fe7fb694-0703-4443-bb4c-398635c11e9d)

In dark mode, the progress bar and surrounding sections feel a bit flattened — it’s hard to visually distinguish the layers (e.g., background vs. container vs. progress track), which impacts readability and hierarchy.

I reviewed the tokens in [maybe-design-system.css](https://github.com/maybe-finance/maybe/blob/main/app/assets/tailwind/maybe-design-system.css) and tried options like bg-surface, bg-overlay, bg-container-inset, etc., but couldn’t find a combination that resolved the contrast issue effectively.

That said, it’s possible I missed a better combination or intended use of those tokens.
Is there a visual reference or Figma spec I could check to understand how these background layers should be applied in cases like this?

Just want to be sure I’m staying within the system and not introducing anything unintended. Appreciate any guidance!

zachgoll commented on app/assets/tailwind/maybe-design-system.css:
```diff
@@ -788,4 +848,12 @@
+  @variant theme-dark {
+    @apply bg-black ;
+  }
+}
```
We can delete this and replace all usages with `bg-container`

zachgoll reviewed (commented):
No worries!  Just a few more little edits and we should be good.

To get those checks passing locally, you can use the following commands:

```
bin/rubocop -f github -a
bundle exec erb_lint ./app/**/*.erb -a
bin/brakeman --no-pager
```

menaguilherme commented:
Hey @zachgoll  – I applied the requested changes and ran the linters you mentioned (`rubocop`, `erb_lint`, `brakeman`). Let me know if anything else is needed!

Thanks again for your time reviewing this 🙏

zachgoll reviewed (approved):
Looks great.  Thanks for your work on this!
